Bottom: 34ced46436e5490105a7fc1812d91a952711b24f
Top:    62262fa37a052e6609aedbce8766eed10853f926
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-03-06 14:59:09 +0000

Speedup for models whilst debugging


---

diff --git a/src/hid/step/assembly.cpp b/src/hid/step/assembly.cpp
index b4de7e2..ddc4eac 100644
--- a/src/hid/step/assembly.cpp
+++ b/src/hid/step/assembly.cpp
@@ -709,7 +709,12 @@ append_model_from_file (Registry *registry,
 //  int file_increment = ( int )( ( ceil( ( max_existing_file_id + 99.0 ) / 1000.0 ) + 1.0 ) * 1000.0 ); /* XXX: RELYING ON SCL NOT CHANGING */
 //  std::cout << "INFO: Expecting a to add " << file_increment << " to entity names" << std::endl;
 
+#if 1
   STEPfile sfile = STEPfile (*registry, *instance_list, "", false);
+#else
+  STEPfile *sfilep = new STEPfile (*registry, *instance_list, "", false);
+  STEPfile &sfile = *sfilep; /* Let us deliberately leak STEPfile for a performance test */
+#endif
 
   // XXX: This appears to throw exceptions from std::ios_base if the file doesn't exist
   try
diff --git a/src/polygon.c b/src/polygon.c
index cb93187..eef7620 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -2474,6 +2474,8 @@ pv_outline_callback (const BoxType * b, void *cl)
     }
   else
     {
+      /* NOT FROM ELEMENT, DROP HOLE */
+      return 1;
       feature_name = STRDUP(pv->Name);
     }
 
diff --git a/src/polygon.h b/src/polygon.h
index 90d0d26..c7be3ea 100644
--- a/src/polygon.h
+++ b/src/polygon.h
@@ -34,8 +34,8 @@
 
 /* Implementation constants */
 
-#define POLY_CIRC_SEGS 40 //24 //20 //8 //40
-#define POLY_CIRC_SEGS_D ((double)POLY_CIRC_SEGS)
+#define POLY_CIRC_SEGS 8
+#define POLY_CIRC_SEGS_F (double)POLY_CIRC_SEGS)
 
 #if 0
 /* THIS IS BROKEN:
